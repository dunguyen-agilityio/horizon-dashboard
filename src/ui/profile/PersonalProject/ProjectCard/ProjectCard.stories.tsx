import type { Meta, StoryObj } from '@storybook/react';

// Components
import ProjectCard from '.';

// Constants
import { NFT_IMAGES } from '@/constants';

const meta = {
  title: 'ui/ProjectCard',
  component: ProjectCard,
  tags: ['autodocs'],
} satisfies Meta<typeof ProjectCard>;

export default meta;
type Story = StoryObj<typeof meta>;

export const ProjectCardWithoutImage: Story = {
  args: {
    id: '4',
    baseProject: 'project #4',
    title: 'Technology behind the Blockchain',
  },
};

export const ProjectCardWithImage: Story = {
  args: {
    ...ProjectCardWithoutImage.args,
    imageProject: NFT_IMAGES.COVER,
  },
};
