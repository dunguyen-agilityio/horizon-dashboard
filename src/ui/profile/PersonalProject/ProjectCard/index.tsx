// Components
import Image from 'next/image';
import { Link } from '@nextui-org/link';
import { Text, BoxIcon } from '@/components';
import { Card, CardBody } from '@nextui-org/card';

// Icons
import { Pen } from '@/icons/';

// Types
import { TEXT_SIZE, TEXT_VARIANT } from '@/types/text';

// Constants
import { NFT_IMAGES } from '@/constants';

export interface IProjectCardProps {
  id?: string;
  title: string;
  baseProject: string;
  imageProject?: string;
}

const ProjectCard = ({
  imageProject = NFT_IMAGES.DEFAULT,
  title,
  baseProject,
}: IProjectCardProps) => (
  <Card className="border-none bg-white dark:bg-indigo-light min-w-[300px] sm:min-w-[420px] xl:min-w-[380px] xl:w-full">
    <CardBody>
      <div className="flex flex-col items-center sm:flex-row justify-between relative">
        <div className="absolute -right-1 -top-1 sm:hidden">
          <BoxIcon icon={<Pen />} />
        </div>
        <div className="relative w-20 h-20 my-2 sm:my-0 rounded-2xl overflow-hidden">
          <Image
            fill
            src={imageProject}
            alt="project-detail"
            className="object-cover"
          />
        </div>
        <div className="mb-2 sm:ml-2 xl:ml-0 sm:mb-0 flex items-center">
          <div className="flex flex-col items-center text-center sm:text-start sm:items-start">
            <Text className="font-medium leading-5">{title}</Text>
            <div className="flex items-center gap-[5px]">
              <Text
                size={TEXT_SIZE.sm}
                variant={TEXT_VARIANT.SECONDARY}
                className="font-medium"
              >
                {baseProject}
              </Text>
              <div className="bg-secondary w-1 h-1 rounded-full" />
              <Link
                href="#"
                underline="always"
                className="text-blue-450 dark:text-white"
              >
                <Text
                  size={TEXT_SIZE.sm}
                  variant={TEXT_VARIANT.QUATERNARY}
                  className="font-medium"
                >
                  See project details
                </Text>
              </Link>
            </div>
          </div>
        </div>
        <div className="hidden sm:block">
          <BoxIcon icon={<Pen />} />
        </div>
      </div>
    </CardBody>
  </Card>
);

export default ProjectCard;
