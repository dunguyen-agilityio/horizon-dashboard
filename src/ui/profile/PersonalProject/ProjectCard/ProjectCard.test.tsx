import { render } from '@testing-library/react';

import ProjectCard from '.';

const titleProjectCard = 'Technology behind the Blockchain';
const refProject = 'project #2';

describe('ProjectCard tests', () => {
  it('Should match snapshot', () => {
    const { container } = render(
      <ProjectCard id="2" title={titleProjectCard} baseProject={refProject} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('Should render icon pencil and correct title', () => {
    const { getAllByTestId, getByText } = render(
      <ProjectCard id="2" title={titleProjectCard} baseProject={refProject} />,
    );
    const listIconPencil = getAllByTestId('pencil-icon');
    expect(listIconPencil[0]).toBeInTheDocument();
    expect(getByText(titleProjectCard)).toBeInTheDocument();
    expect(getByText(refProject)).toBeInTheDocument();
  });
});
