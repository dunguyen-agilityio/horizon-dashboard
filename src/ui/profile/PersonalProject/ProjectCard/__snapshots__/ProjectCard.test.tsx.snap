// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCard tests Should match snapshot 1`] = `
<div>
  <div
    class="flex flex-col relative overflow-hidden h-auto text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 shadow-medium rounded-large transition-transform-background motion-reduce:transition-none border-none bg-white dark:bg-indigo-light min-w-[300px] sm:min-w-[420px] xl:min-w-[380px] xl:w-full"
    tabindex="-1"
  >
    <div
      class="relative flex w-full p-3 flex-auto flex-col place-content-inherit align-items-inherit h-auto break-words text-left overflow-y-auto subpixel-antialiased"
    >
      <div
        class="flex flex-col items-center sm:flex-row justify-between relative"
      >
        <div
          class="absolute -right-1 -top-1 sm:hidden"
        >
          <i
            class="w-6 h-6 flex justify-center items-center hover:cursor-pointer fill-secondary dark:fill-white"
          >
            <svg
              data-testid="pencil-icon"
              fill="none"
              height="17"
              viewBox="0 0 17 17"
              width="17"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#clip0_101_11790)"
              >
                <path
                  d="M2.125 12.3675V14.5208C2.125 14.7192 2.28083 14.875 2.47917 14.875H4.6325C4.72458 14.875 4.81667 14.8396 4.88042 14.7688L12.6154 7.04083L9.95917 4.38458L2.23125 12.1125C2.16042 12.1833 2.125 12.2683 2.125 12.3675ZM14.6696 4.98667C14.9458 4.71042 14.9458 4.26417 14.6696 3.98792L13.0121 2.33042C12.7358 2.05417 12.2896 2.05417 12.0133 2.33042L10.7171 3.62667L13.3733 6.28292L14.6696 4.98667Z"
                  fill="#A3AED0"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_101_11790"
                >
                  <rect
                    fill="white"
                    height="17"
                    width="17"
                  />
                </clippath>
              </defs>
            </svg>
          </i>
        </div>
        <div
          class="relative w-20 h-20 my-2 sm:my-0 rounded-2xl overflow-hidden"
        >
          <img
            alt="project-detail"
            class="object-cover"
            data-nimg="fill"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=%2Fnft-default.webp&w=3840&q=100"
            srcset="/_next/image?url=%2Fnft-default.webp&w=640&q=100 640w, /_next/image?url=%2Fnft-default.webp&w=750&q=100 750w, /_next/image?url=%2Fnft-default.webp&w=828&q=100 828w, /_next/image?url=%2Fnft-default.webp&w=1080&q=100 1080w, /_next/image?url=%2Fnft-default.webp&w=1200&q=100 1200w, /_next/image?url=%2Fnft-default.webp&w=1920&q=100 1920w, /_next/image?url=%2Fnft-default.webp&w=2048&q=100 2048w, /_next/image?url=%2Fnft-default.webp&w=3840&q=100 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
          />
        </div>
        <div
          class="mb-2 sm:ml-2 xl:ml-0 sm:mb-0 flex items-center"
        >
          <div
            class="flex flex-col items-center text-center sm:text-start sm:items-start"
          >
            <p
              class="text-primary dark:text-white text-medium font-medium leading-5"
            >
              Technology behind the Blockchain
            </p>
            <div
              class="flex items-center gap-[5px]"
            >
              <p
                class="text-secondary dark:text-white text-small font-medium"
              >
                project #2
              </p>
              <div
                class="bg-secondary w-1 h-1 rounded-full"
              />
              <a
                class="relative inline-flex items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium underline hover:opacity-80 active:opacity-disabled transition-opacity underline-offset-4 text-blue-450 dark:text-white"
                href="#"
                role="link"
                tabindex="0"
              >
                <p
                  class="text-blue-450 dark:text-white text-small font-medium"
                >
                  See project details
                </p>
              </a>
            </div>
          </div>
        </div>
        <div
          class="hidden sm:block"
        >
          <i
            class="w-6 h-6 flex justify-center items-center hover:cursor-pointer fill-secondary dark:fill-white"
          >
            <svg
              data-testid="pencil-icon"
              fill="none"
              height="17"
              viewBox="0 0 17 17"
              width="17"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#clip0_101_11790)"
              >
                <path
                  d="M2.125 12.3675V14.5208C2.125 14.7192 2.28083 14.875 2.47917 14.875H4.6325C4.72458 14.875 4.81667 14.8396 4.88042 14.7688L12.6154 7.04083L9.95917 4.38458L2.23125 12.1125C2.16042 12.1833 2.125 12.2683 2.125 12.3675ZM14.6696 4.98667C14.9458 4.71042 14.9458 4.26417 14.6696 3.98792L13.0121 2.33042C12.7358 2.05417 12.2896 2.05417 12.0133 2.33042L10.7171 3.62667L13.3733 6.28292L14.6696 4.98667Z"
                  fill="#A3AED0"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_101_11790"
                >
                  <rect
                    fill="white"
                    height="17"
                    width="17"
                  />
                </clippath>
              </defs>
            </svg>
          </i>
        </div>
      </div>
    </div>
  </div>
</div>
`;
